### ISSUE TITLE
Rule Change: Allow whitespace in between curly braces with empty bodies

### ISSUE THREAD

**Issue by nolawnchairs** [eslint/eslint#16963]
### What rule do you want to change?

object-curly-spacing

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new option

### Example code

```js
// Desired
const emptyObject = { }

// Actual
const emptyObject = {}
```

### What does the rule currently do for this code?

The rule currently only allows for space in populated object literals, function and constructor bodies, but enforces only NO space for empty objects.

### What will the rule do after it's changed?

Allow a space between empty object literals, constructor and function bodies without implementation:

```ts
constructor() { }

function thing() { }

const emptyObject = { }

```

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

My editor somehow automatically does this on file save, but this rule will remove all whitespace if others who edit the code don't have the same editor setting. An ESLint rule fix would allow this.
